server/routes/api/comments/comments.ts

```diff
@@ -61,18 +62,53 @@ router.post(
   checkCommentingEnabled(),
   validate(T.CollectionsListSchema),
   async (ctx: APIContext<T.CollectionsListReq>) => {
-    const { sort, direction, documentId } = ctx.input.body;
+    const { sort, direction, documentId, collectionId } = ctx.input.body;
     const { user } = ctx.state.auth;
 
-    const document = await Document.findByPk(documentId, { userId: user.id });
-    authorize(user, "read", document);
-
-    const comments = await Comment.findAll({
-      where: { documentId },
-      order: [[sort, direction]],
-      offset: ctx.state.pagination.offset,
-      limit: ctx.state.pagination.limit,
-    });
+    let comments;
+    if (documentId) {
+      const document = await Document.findByPk(documentId, { userId: user.id });
+      authorize(user, "read", document);
+      comments = await Comment.findAll({
+        where: { documentId },
```

Comment:
```suggestion
        where: {
          teamId: user.teamId,
          documentId: document.id,
        },
```

Lets use `document.id` here, it will allow `documentId` to be in the slug format and still work.

Reply:
On it. Also, do you want the `teamId` condition as well here ( as suggested in your suggestion ). Asking because `Comment` does not have teamId so we'll need to join with the `Document` to filter by the teamId. Previously also, there was no teamId filter.

Reply:
Ah, good point you can leave it off.